.github/workflows/release.yml

```diff
@@ -0,0 +1,42 @@
+name: Release Alpha
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_version:
+        description: 'Release version (e.g. 0.1.0, 2.1.1, ...)'
+        required: true
+      next_version:
+        description: 'Next version (e.g. 0.2.0, 2.2.0, ...)'
+        required: true
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout Alpha  
+    # See https://github.com/marketplace/actions/checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Validate Gradle Wrapper Checksum      
+    # See https://github.com/marketplace/actions/gradle-wrapper-validation
+      uses: gradle/wrapper-validation-action@v1
+    - name: Install Java
+    # See https://github.com/marketplace/actions/setup-java-jdk  
+      uses: actions/setup-java@v3
+      with:
+        java-version: 11
+        distribution: 'temurin'
+    - name: Prepare git config
+      run: |
+        git config --local user.name alpha-release-action
+        git config --local user.email alpha-release-action@no.where    
+    - name: Run Gradle Release Plugin
+    # See https://github.com/marketplace/actions/gradle-build-action
+    # https://github.com/researchgate/gradle-release  
+      uses: gradle/gradle-build-action@v2
+      with:
+        arguments: release -Prelease.useAutomaticVersion=true -Prelease.releaseVersion=${{ github.event.inputs.release_version}} -Prelease.newVersion=${{ github.event.inputs.next_version}}
```

Comment:
If there is an easy way to make sure that release versions are increasing only, I'd like to see that used. But I guess that is really hard to achieve in cases where, e.g., a new patch is back-ported to an older version. Is there some way to exclude that version numbers are duplicated?

Reply:
See above comments about my failed attempt to do that via script.
Log story short, I think it's more trouble than it's worth: Since you may have perfectly valid reasons to release pretty much any version at any point in time, I don't think such a check could ever be "smart enough" to be useful. Even if your current development version (i.e. version in `master`) were `2.3.0`, you could still find a bug in version `1.7.0` that needs urgent fixing but is not easily ported to your current version, thus requiring you to release a version `1.7.1` that branches from your old tag.
Virtually the only useful check on release versions is if you already have a tag for the same version - and that happens automatically since git won't let you create duplicate tags.